File: app/src/main/java/com/example/myapplication/TransaksiAdapter.java
package com.example.myapplication;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class TransaksiAdapter extends RecyclerView.Adapter<TransaksiAdapter.ViewHolder> {

    private List<Transaksi> transaksiList;

    public TransaksiAdapter(List<Transaksi> transaksiList) {
        this.transaksiList = transaksiList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_transaksi, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Transaksi item = transaksiList.get(position);
        holder.tvTanggal.setText(item.getTanggal());
        holder.tvDeskripsi.setText(item.getDeskripsi());
        holder.tvPoin.setText(item.getPoin());

        // Ubah warna berdasarkan tipe transaksi
        if (item.isMasuk()) {
            holder.tvPoin.setTextColor(Color.parseColor("#4CAF50")); // Hijau
        } else {
            holder.tvPoin.setTextColor(Color.parseColor("#F44336")); // Merah
        }
    }

    @Override
    public int getItemCount() {
        return transaksiList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTanggal, tvDeskripsi, tvPoin;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvTanggal = itemView.findViewById(R.id.tvTanggal);
            tvDeskripsi = itemView.findViewById(R.id.tvDeskripsi);
            tvPoin = itemView.findViewById(R.id.tvPoin);
        }
    }
}

File: app/src/main/java/com/example/myapplication/TransaksiAdapter.kt
package com.example.myapplication

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class TransaksiAdapter(
    private val transaksiList: List<Transaksi>
) : RecyclerView.Adapter<TransaksiAdapter.ViewHolder>() {

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvTanggal: TextView = itemView.findViewById(R.id.tvTanggal)
        private val tvDeskripsi: TextView = itemView.findViewById(R.id.tvDeskripsi)
        private val tvPoin: TextView = itemView.findViewById(R.id.tvPoin)

        fun bind(transaksi: Transaksi) {
            tvTanggal.text = transaksi.tanggal
            tvDeskripsi.text = transaksi.deskripsi
            tvPoin.text = transaksi.poin

            // Ubah warna berdasarkan tipe transaksi
            if (transaksi.isMasuk) {
                tvPoin.setTextColor(Color.parseColor("#4CAF50")) // Hijau
            } else {
                tvPoin.setTextColor(Color.parseColor("#F44336")) // Merah
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_transaksi, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(transaksiList[position])
    }

    override fun getItemCount(): Int = transaksiList.size
}
